cob_sick_s300/ros/src/cob_sick_s300.cpp

```diff
@@ -220,9 +220,10 @@ class NodeClass
 		// other function declarations
 		void publishLaserScan(std::vector<double> vdDistM, std::vector<double> vdAngRAD, std::vector<double> vdIntensAU, unsigned int iSickTimeStamp, unsigned int iSickNow)
 		{
-			if(ros::Time::now()-loop_rate_.now()>=ros::Duration(1./publish_frequency))
```

Comment:
I think we should set a variable
```
ros::Time now = ros::Time::now()
```
right at the beginning of the publish function  and then use `now` throughout the function imstead of getting new times everytime